device.go

```diff
@@ -158,11 +166,55 @@ func newBlockDevice(devInfo DeviceInfo) *BlockDevice {
 	}
 }
 
-func (device *BlockDevice) attach(h hypervisor) error {
+func (device *BlockDevice) attach(h hypervisor, c *Container, create bool) error {
+	device.DeviceInfo.ID = uuid.Generate().String()
+	devID := fmt.Sprintf("drive-%s", device.DeviceInfo.ID)
```

Comment:
I know it's trivial, but to guarantee this doesn't diverge from the format in `detach()`, I'd be tempted to create something like:

```
func makeDevIDForHypervisor(deviceID string) string {
    return fmt.Sprintf("drive-%s", deviceID)
}
```